Source/Core/DolphinQt2/Config/GameConfigWidget.cpp

```diff
@@ -0,0 +1,412 @@
+
+  connect(m_deterministic_dual_core,
+          static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged), this,
+          &GameConfigWidget::SaveSettings);
+  connect(m_depth_slider, static_cast<void (QSlider::*)(int)>(&QSlider::valueChanged), this,
+          &GameConfigWidget::SaveSettings);
+  connect(m_convergence_spin, static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged), this,
+          &GameConfigWidget::SaveSettings);
+}
+
+void GameConfigWidget::LoadCheckBox(QCheckBox* checkbox, const std::string& section,
+                                    const std::string& key)
+{
+  bool checked;
+
+  if (m_gameini_local.GetOrCreateSection(section)->Get(key, &checked))
+    return checkbox->setCheckState(checked ? Qt::Checked : Qt::Unchecked);
+  else if (m_gameini_default.GetOrCreateSection(section)->Get(key, &checked))
+    return checkbox->setCheckState(checked ? Qt::Checked : Qt::Unchecked);
+  else
+    checkbox->setCheckState(Qt::PartiallyChecked);
+}
+
+void GameConfigWidget::SaveCheckBox(QCheckBox* checkbox, const std::string& section,
+                                    const std::string& key)
+{
+  // Delete any existing entries from the local gameini if checkbox is undetermined.
+  // Otherwise, write the current value to the local gameini if the value differs from the default
+  // gameini values.
+  // Delete any existing entry from the local gameini if the value does not differ from the default
+  // gameini value.
+
+  if (checkbox->checkState() == Qt::PartiallyChecked)
+  {
+    m_gameini_local.DeleteKey(section, key);
+    return;
+  }
+
+  bool checked = checkbox->checkState() == Qt::Checked;
+
+  if (m_gameini_default.Exists(section, key))
+  {
+    bool default_value;
+    m_gameini_default.GetOrCreateSection(section)->Get(key, &default_value);
+
+    if (default_value != checked)
+      m_gameini_local.GetOrCreateSection(section)->Set(key, checked);
+    else
+      m_gameini_local.DeleteKey(section, key);
+
+    return;
+  }
+
+  m_gameini_local.GetOrCreateSection(section)->Set(key, checked);
+
+  return;
+}
+
+void GameConfigWidget::LoadSettings()
+{
+  // Load state information
+  m_state_combo->setCurrentIndex(m_game.GetRating());
+  m_state_comment_edit->setText(m_game.GetIssues());
+
+  // Load game-specific settings
+
+  // Core
+  LoadCheckBox(m_enable_dual_core, "Core", "CPUThread");
+  LoadCheckBox(m_enable_mmu, "Core", "MMU");
+  LoadCheckBox(m_skip_dcbz, "Core", "DCBZ");
+  LoadCheckBox(m_enable_fprf, "Core", "FPRF");
+  LoadCheckBox(m_sync_gpu, "Core", "SyncGPU");
+  LoadCheckBox(m_enable_fast_disc, "Core", "FastDiscSpeed");
+  LoadCheckBox(m_use_dsp_hle, "Core", "DSPHLE");
+
+  std::string determinism_mode;
+  if (!m_gameini_local.GetIfExists("Core", "GPUDeterminismMode", &determinism_mode))
+    if (!m_gameini_default.GetIfExists("Core", "GPUDeterminismMode", &determinism_mode))
+      determinism_mode = DETERMINISM_NOT_SET_STRING;
+
+  if (determinism_mode == DETERMINISM_AUTO_STRING)
```

Comment:
IMO, a switch statement would be better here.

Reply:
A switch on a string?

Reply:
Oh whoops, disregard what I said.